tests/postgres_tests/test_operations.py

```diff
@@ -289,6 +294,34 @@ def test_allow_migrate(self):
         self.assertEqual(len(captured_queries), 2)
         self.assertIn("DROP EXTENSION IF EXISTS", captured_queries[1]["sql"])
 
+    @override_settings(DATABASE_ROUTERS=[MigrateWhenHinted()])
+    def test_allow_migrate_based_on_hints(self):
+        operation_no_hints = CreateExtension("tablefunc")
+        self.assertEqual(operation_no_hints.hints, {})
+
+        operation_hints = CreateExtension("tablefunc", hints={"a_hint": True})
+        self.assertEqual(operation_hints.hints, {"a_hint": True})
+
+        project_state = ProjectState()
+        new_state = project_state.clone()
+
+        with CaptureQueriesContext(connection) as captured_queries:
+            with connection.schema_editor(atomic=False) as editor:
+                operation_no_hints.database_forwards(
+                    self.app_label, editor, project_state, new_state
+                )
+
+        self.assertEqual(len(captured_queries), 0)
+
+        with CaptureQueriesContext(connection) as captured_queries:
+            with connection.schema_editor(atomic=False) as editor:
+                operation_hints.database_forwards(
+                    self.app_label, editor, project_state, new_state
+                )
```

Comment:
```suggestion
        with (
            CaptureQueriesContext(connection) as captured_queries,
            connection.schema_editor(atomic=False) as editor,
        ):
            operation_hints.database_forwards(
                self.app_label, editor, project_state, new_state
            )
```

Reply:
should I update the test I copied this from as well?